### DISCUSSION TITLE
How to get stremlink for https://24vod-adaptive.akamaized.net/ url

### DISCUSSION THREAD

**Discussion by thesilvia** [streamlink/streamlink#5644] [Q&A] [answered]
### Description

Hi guys
I am completely new in this section . I want to know is it possible to get direct hls link from the follow link

https://24vod-adaptive.akamaized.net/exp=1698697569~acl=%2F5cceeec1-ebe8-460e-992d-bece28d2d9a7%2F%2A~hmac=6b821e8ddbe761b7099c6b086f9688305838417a7cecc841709d9b4b94695b3d/5cceeec1-ebe8-460e-992d-bece28d2d9a7/parcel/audio/b3d493b4.mp4?r=dXM%3D&range=6650-19201
 

What should I put in ???

streamlink “???” best --hls-segment-threads 5 -o "D:/test.ts"

Thanks in advance
![Screenshot 2023-10-30 132303](https://github.com/streamlink/streamlink/assets/71188838/97fa13e3-a37f-47d8-a35b-35524c6587cb)

**Comment by bastimeyer** [comment]
The link you've posted is not an HLS stream, as you can see in the HTTP response headers.

```
$ curl -Is 'https://24vod-adaptive.akamaized.net/exp=1698697569~acl=%2F5cceeec1-ebe8-460e-992d-bece28d2d9a7%2F%2A~hmac=6b821e8ddbe761b7099c6b086f9688305838417a7cecc841709d9b4b94695b3d/5cceeec1-ebe8-460e-992d-bece28d2d9a7/parcel/audio/b3d493b4.mp4?r=dXM%3D&range=6650-19201' | grep Content-Type:
Content-Type: video/mp4
```

An what's shown on the screenshot you've posted is some kind of segmented download of an mp4 container, but this could be anything, as you didn't post the actual link to the website.

**Reply by thesilvia** [reply]
Thank you for the explanation. Would you please kindly walk me through getting the HLS stream link for the video please?

**Reply by thesilvia** [reply]
is this a HLS stream link?
https://app.kartra.com/external_video/vimeo/875544068?h=eb22c80288&amp

This still does not work

C:\Program Files (x86)\Streamlink\bin>streamlink "https://app.kartra.com/external_video/vimeo/875544068.m3u8" best --hls-segment-threads 5 -o "D:/test.ts"
[cli][info] Found matching plugin hls for URL https://app.kartra.com/external_video/vimeo/875544068.m3u8
[stream.hls_playlist][warning] Malformed HLS Playlist. Expected #EXTM3U, but got <!DOCTYPE html>
error: Failed to parse playlist: Missing #EXTM3U header

**Reply by bastimeyer** [reply, marked as answer]
> [pssummit.com/DayOnePVS5](https://www.pssummit.com/DayOnePVS5)
> [app.kartra.com/external_video/vimeo/875544068?h=eb22c80288&](https://app.kartra.com/external_video/vimeo/875544068?h=eb22c80288&amp)

This appears to be a custom JSON-based DASH implementation which downloads the segmented mp4 container by setting the `range` query parameter and which reads the init segment from embedded data in the JSON payload (`master.json`) encoded as base64.

Streamlink does not support that and you'll have to figure this out on your own. Try to remove the `range` parameter and see if you can download the mp4 container as a whole. But as said, no support for any of that here. Good luck.

**Reply by bastimeyer** [reply]
Since there's a vimeo reference in the URL, see if you can find the video on vimeo. Streamlink does have a plugin for that.

**Comment by thesilvia** [comment]
Thanks Bastimeyer. This is the site and I would like to download the video from it:
https://www.pssummit.com/DayOnePVS5

**Comment by JakeStorm-Dev** [comment]
Not sure if you're still hunting down how to do this but I can shed a little insight on how.  Providing you're not wanting to stream directly more permanently, this can be done by using VLC media player to remux into a download.  

1) First you need to isolate the stream.  In your browser, open the page from the link.  Then open developer tools, then "Network" tab.  Then refresh the page in your broswer to obtain a clean look at the details from start.  Next scroll down to find the stream.  In this case, the first video on the screen shows up as: "876070104?h=2466922a89" with the header of "https:// player.vimeo.com/video/876070104?h=2466922a89".  This is the stream you're looking for.  So click on it and open the "Response" tab .

2) Next we need to see what CDN Server it's coming from.  Note that Vimeo loves obscuring their files and even throws out blank feeds or "off the air" feeds to do so.  But ideally, for download, you only need to get one working link over to VLC so that it can mux the rest into a working MP4.  Also of note should be the various CDN prefixes ( "i.vimeocdn" = image and  "f.vimeo" =file) - So in this example, the first video link we're looking for should be located in the f.vimeo section of the response.  Here we see the following link as the json file previously mentioned but no direct reference to the m3u8:

`https://13vod-adaptive.akamaized.net/exp=1714513826~acl=%2Fa8ca4874-358e-476a-8ac1-9bc8ae24540c%2F%2A~hmac=1b33f1b4fe72466bc2c3107a21771cecc41fabaaae22910de350cad4dc8c745a/a8ca4874-358e-476a-8ac1-9bc8ae24540c/sep/video/10824f70,657b5a1e,79ab7c64,b11dc595,e4bb87d9/audio/a4d02b00,b93a9503,ba7312ae/master.json?base64_init=1\u0026query_string_ranges=1`

3) This is where a bit of understanding of IT organizational skills comes into play.  Because much of the coding online these days is in relational databases, filenames tend to mirror each other for functionality and ease of coding.  So in this instance the json file in question also bears the same filename as our m3u8.  So a strong bit of guess work leads us to the m3u8 by simply truncating the json's parameters and the json extension and apply the m3u8 exension as follows:

`https://13vod-adaptive.akamaized.net/exp=1714513826~acl=%2Fa8ca4874-358e-476a-8ac1-9bc8ae24540c%2F%2A~hmac=1b33f1b4fe72466bc2c3107a21771cecc41fabaaae22910de350cad4dc8c745a/a8ca4874-358e-476a-8ac1-9bc8ae24540c/sep/video/10824f70,657b5a1e,79ab7c64,b11dc595,e4bb87d9/audio/a4d02b00,b93a9503,ba7312ae/master.m3u8`

4) From here we can take our newly created m3u8 link and open it in VLC using the "Open Network Stream" function.  It should have no issue playing in this manner.  Keep in mind that this link has been run through Akamai and is encrypted with and expiry. 
 That's what the "exp=" is in the link followed by the date and time stamp it was created.  Generally Akamai links run from 4 to 24 hours, but should be enough time in which "record the content" for personal use if you so choose.  As to how to rip the video from the network stream, that starts to dive into the grey area of internet ethics and I'll not be covering that here.  The point being that legally and ethically, if content is offered for free, you can view it however you wish.  In this example we've not bypassed any encryption nor hacked any servers; only changed the method (or device) in which we view it on.  From the legal sense, this won't get you into trouble.  But the same laws governing VHS, CD's,  and DVD's apply as to if you're going to "record" content for personal use.  Just keep in mind distribution requires license on content you do not own.

Hope this helps :)
